marshmallow_recipe/bake.py

```diff
@@ -183,7 +183,9 @@ def pre_load(self, data: dict[str, Any]) -> Any:
 def _get_field_default(field: dataclasses.Field[_T]) -> Any:
     default_factory = field.default_factory
     if default_factory is not dataclasses.MISSING:  # type: ignore
-        raise ValueError(f"Default factory is not supported for {field}")
+        origin = typing_inspect.get_origin(field.type) or field.type
+        if origin in (list, List, dict, Dict) or dataclasses.is_dataclass(field.type):
```

Comment:
Why such a restriction? Why can't we support any type of default factories?

Reply:
We can. But this case is out of the scope of this PR

Reply:
The thing is that the support for any types is the same as for these limited types (you don't have any list/dict-specific code anywhere), thus this restriction is totally artificial and unnecessary

Reply:
There are. Defaults are not supported for lists/dicts/nested

Reply:
Let's distinct simple and complex types, please?

Reply:
Why?

Reply:
Just because complex types are our future. But we have usecase for simple types right now

Reply:
Anyway, I am not fond of having this checks here. Previously, all checks about default values were made in *_field function, now this logic is in multiple places.

Reply:
Also I have checked and it is impossible to create a dataclass with list/dict fields having mutable default.

Reply:
I will push fixes soon.

Reply:
```
  File "/Users/yurypliner/.pyenv/versions/3.10.2/lib/python3.10/dataclasses.py", line 1185, in dataclass
    return wrap(cls)
  File "/Users/yurypliner/.pyenv/versions/3.10.2/lib/python3.10/dataclasses.py", line 1176, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
  File "/Users/yurypliner/.pyenv/versions/3.10.2/lib/python3.10/dataclasses.py", line 956, in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
  File "/Users/yurypliner/.pyenv/versions/3.10.2/lib/python3.10/dataclasses.py", line 813, in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
ValueError: mutable default <class 'list'> for field a is not allowed: use default_factory
```

<img width="703" alt="image" src="https://user-images.githubusercontent.com/664889/210790567-a79fe0c5-1090-45c4-bc9a-958e04232f9a.png">